components/Chips/src/MDCAlignedCollectionViewFlowLayout.h

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+typedef enum : NSUInteger {
+  MDCCollectionHorizontalAlignmentLeading,
+  MDCCollectionHorizontalAlignmentTrailing,
+  MDCCollectionHorizontalAlignmentLeft,
+  MDCCollectionHorizontalAlignmentRight
+} MDCCollectionHorizontalAlignment;
+
+/*
+ MDCAlignedCollectionViewFlowLayout is an aligned version of UICollectionViewFlowLayout.
```

Comment:
I wasn't sure what "aligned" meant at first. Add a second sentence "UICollectionViewFlowLayout only provides justified alignment of its cells, MDCAlignedCollectionViewFlowLayout adds flush left and flush right alignments."? https://en.wikipedia.org/wiki/Typographic_alignment

Might consider renaming to MDCFlushAligned*, but that's a nit.

Reply:
This isn't necessarily correct:

> UICollectionViewFlowLayout only provides justified alignment of its cells

By default flow layout is flush left.

Reply:
Jeff that isn't correct. UICollectionViewFlowLayout does not left align cells. It only provides a justified layout. Hence the need for this class.

Reply:
Hah! Well I'll be darned - I'd always read `minimumInteritemSpacing` as simply `iteritemSpacing`. Is showing how long since I've used a flow layout from scratch haha.

![tumblr_inline_orc8s2sita1tlruqr_540](https://user-images.githubusercontent.com/45670/31726960-6079ea70-b3f6-11e7-9364-32e56111e71c.gif)